src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -477,6 +477,20 @@ object PruneDeadFields {
         memoizeMatrixIR(child, dep, memo)
       case MatrixUnionRows(children) =>
         children.foreach(memoizeMatrixIR(_, requestedType, memo))
+      case UnlocalizeEntries(rowsEntries, cols, fieldName) =>
+        val m = Map(MatrixType.entriesIdentifier -> fieldName)
+        val minRowEntries = minimal(rowsEntries.typ)
+        val rowsEntriesDep = minRowEntries.copy(
+          globalType = requestedType.globalType,
+          rowType = unify(rowsEntries.typ.rowType, minRowEntries.rowType, requestedType.rvRowType.rename(m))
+        )
+        memoizeTableIR(rowsEntries, rowsEntriesDep, memo)
+
+        val minCols = minimal(cols.typ)
+        val colDep = cols.typ.copy(
+          rowType = unify(cols.typ.rowType, minCols.rowType, requestedType.colType),
+          globalType = TStruct())
```

Comment:
we still have some existing datasets written with a required empty struct for global type. This will crash on those datasets.

Should be `minCols.globalType`

Reply:
The intention is to throw away the `cols` globals, Will doing what is suggested indicate that we won't need them?

Reply:
it will, but if the globals were previously a `+TStruct()` (required TStruct) this could cause an assertion error.

Reply:
minCols.globalType will always be either `TStruct()` or `+TStruct()`, requiredness matching the original struct.